### PR TITLE
tag sort dedup

### PR DESCRIPTION
**Pull request by extemporalgenome**
This is meant to be read commit-by-commit.

- tag.go: conventional test naming.
- add golang.org/x/exp/slices dependency
- tag.go: use slices package Sort functionality
- SortTags deduplicates, uses slices.SortStable

### CODE REVIEW

File: tag.go [hunk 1]
```diff
@@ -47,21 +47,54 @@ func SortTags(tags []Tag) []Tag {
...
+		// No input tags had non-empty names:
+		// return nil to be consistent for ease of testing.
+		return nil
+	}
 
-func concatTags(t1 []Tag, t2 []Tag) []Tag {
+	return out
+}
+
+// mergeTags returns the sorted, deduplicated-by-name union of t1 and t2.
```
Comment (kevinburkesegment) [inline tag.go:86]: Can you explain in the comment what strategy we use to deduplicate (last write wins).

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
 
-    - name: Download Dependencies
-      run: go mod download
+      - name: Identify OS
+        run: uname -a
 
-    - name: Run Tests
-      run: go test -race -tags=${{ matrix.tags }} ./...
+      - name: Identify Go Version
+        run: go version
```
Comment (kevinburkesegment) [inline .github/workflows/go.yml:31]: These start a new container per command so there's a fair amount of overhead. How about e.g.

```yml
name: Identify Host 
run: |
    uname -a
    go version
```

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -15,13 +16,13 @@ jobs:
...
-          go-version: ^1.18
+          go-version: '>=1.20'
 
       - uses: actions/checkout@v3
 
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3.1.0
         with:
-          version: v1.45.2
+          version: v1.53
```
Comment (kevinburkesegment) [inline .github/workflows/golangci-lint.yml:28]: Thanks for updating this.

**Review by kevinburkesegment — APPROVED** [review]

File: .github/workflows/go.yml (hunk 2 above)
Comment (extemporalgenome) [reply .github/workflows/go.yml:31]: For BuildKite that'd definitely be the case, but for Github Actions, it looks like jobs create their own container/VM/whatever, but steps are just separate OS processes in that same container.

As such, steps look like they're no more expensive than running separate commands in a single shell. By making them separate steps, they do create nice UI sections though, and the above info was useful for debugging the especially stale workflow config.

File: tag.go (hunk 1 above)
Comment (extemporalgenome) [reply tag.go:87]: The latest commit has hopefully addressed this. Please re-review.

**Review by kevinburkesegment — APPROVED** [review]

**Pull request merged**